src/ByPropertyIdArray.php

```diff
@@ -52,7 +53,7 @@ class ByPropertyIdArray extends ArrayObject {
 	/**
```

Comment:
It's an array of arrays of PropertyIdProvider. A possible type hint for that would be `PropertyIdProvider[][]`, but since not all tools understand that `array[]` is correct.